go: check for NaN in lt, lte, gt, gte

torkelrogstad opened the pull request:
Closes #548 

Don't know if this is supposed to have tests or not. Not really familiar with Bazel, so would need some pointers as to where to proceed on that.

torkelrogstad commented:
@rodaine would it be possible to get some eyes on this?

I believe the CI failures are not related. If so, I think this is something that fixes a bug quite a few people would stumble upon (I've done so, at least)

torkelrogstad commented:
Ping @rmichela 

:pray:

torkelrogstad commented:
This is not stale, please remove the tag.

torkelrogstad commented:
Rebased.

torkelrogstad commented:
@htuch is it possible to get some eyes on this?

htuch commented:
I'm not an active maintainer, I think @rmichela @lizan @snowp might be better folks to take a look. FWIW it seems totally reasonable to me, but we would definitely want to see tests for this behavior added.

torkelrogstad commented:
Thanks for replying  @htuch. How would I get started on tests? Do you for example do a golden file setup where the result from the templates are compared against a known good result?

htuch commented:
Yes, there are places for declaring test cases, e.g. for string const constraint:
* https://github.com/envoyproxy/protoc-gen-validate/blob/main/tests/harness/cases/strings.proto#L8
* https://github.com/envoyproxy/protoc-gen-validate/blob/dc51e59d2e463996ef24e45621cf61f95fa64b0f/tests/harness/executor/cases.go#L739

You should be able to fit this into the relevant float tests in these directories.

ElliotMJackson commented:
relates to #85

ElliotMJackson reviewed (changes requested):
happy to get this in, would like to see it put through the test harness

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/bufbuild/protoc-gen-validate?pullRequest=561) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/bufbuild/protoc-gen-validate?pullRequest=561) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/bufbuild/protoc-gen-validate?pullRequest=561) it.</sub>

torkelrogstad commented:
Excited to see new maintainers here @ElliotMJackson. I rebased this on newest main branch, and added what I think are tests? Can't get them to run locally, let's see what happens on CI.

ElliotMJackson commented:
We are excited to be here @torkelrogstad! I've kicked off CI 🤞 

for future reference, run them locally with a `make ci` or a safer space is in the container:

```shell
docker build -t protoc-gen-validate .
docker run --rm protoc-gen-validate ci
```

torkelrogstad commented:
Found some issues here I'm working on fixing, pushing up later

elliotmjackson commented:
closing due to inactivity
